File: 3143-longest-unequal-adjacent-groups-subsequence-i/3143-longest-unequal-adjacent-groups-subsequence-i.py
class Solution(object):
    def getLongestSubsequence(self, words, groups):
        n=len(words)
        v=[words[0]]
        for i in range(1,n,+1):
            if groups[i]!=groups[i-1]:
                v.append(words[i])
        return v

File: 3143-longest-unequal-adjacent-groups-subsequence-i/3143-longest-unequal-adjacent-groups-subsequence-i.cpp
class Solution {
public:
    vector<string> getLongestSubsequence(vector<string>& words, vector<int>& groups) {
        int n=words.size();
        vector<string>v;
        v.push_back(words[0]);
        for(int i=1;i<n;i++){
            if(groups[i]!=groups[i-1]) v.push_back(words[i]);
        }
        return v;
    }
};